include/librustzcash.h

```diff
@@ -270,6 +270,35 @@ extern "C" {
         uint64_t vpub_old,
         uint64_t vpub_new
     );
+
+    /// Derive the master ExtendedSpendingKey from a seed.
+    void librustzcash_zip32_xsk_master(
+        const unsigned char *seed,
+        size_t seedlen,
+        unsigned char *xsk_master
+    );
+
+    /// Derive a child ExtendedSpendingKey from a parent.
+    void librustzcash_zip32_xsk_derive(
+        const unsigned char *xsk_parent,
+        uint32_t i,
+        unsigned char *xsk_i
+    );
+
+    /// Derive a child ExtendedFullViewingKey from a parent.
+    bool librustzcash_zip32_xfvk_derive(
+        const unsigned char *xfvk_parent,
+        uint32_t i,
+        unsigned char *xfvk_i
+    );
+
+    /// Derive a PaymentAddress from an ExtendedFullViewingKey.
```

Comment:
I'm wondering how this will be designed to work without repeating addresses. E.g. if `j_ret=j+1` we should store `j_ret` somewhere to know we should start from `j+2` for the next address.

Reply:
Correct; you are given back the "index" of the address being used, and should store it so that you know to try `index + 1` the next time you want an address.

Reply:
Wondering if the following interface is better: You give as input the index j of a valid diversifier - the method return an error if j is *not* a valid index; and if it is returns the *next* valid diversifier

Reply:
It might be; we can explore that in a follow-up PR when we have more users of the API.